clang/include/clang/3C/ConstraintVariables.h

```diff
@@ -261,6 +268,18 @@ class PointerVariableConstraint : public ConstraintVariable {
   // pointers.
   bool IsZeroWidthArray;
 
+  // Was this variable a checked pointer in the input program?
+  // This is important for two reasons: (1) externs that are checked should be
+  // kept that way during solving, (2) nothing that was originally checked
+  // should be modified during rewriting.
+  bool OriginallyChecked;
```

Comment:
The goal of preserving Checkedness of a type that's in the input is a general goal. Will this flag affect rewriting of pointer types that are *not* typedefs? I.e., was it possible that checked types were previously getting rewritten prior to you making this change?